test/compute/2019-12-01/armcompute/zz_generated_availabilitysets_client.go

```diff
@@ -35,17 +35,17 @@ type AvailabilitySetsClient struct {
 // credential - used to authorize requests. Usually a credential from azidentity.
 // options - pass nil to accept the default values.
 func NewAvailabilitySetsClient(subscriptionID string, credential azcore.TokenCredential, options *arm.ClientOptions) *AvailabilitySetsClient {
-	cp := arm.ClientOptions{}
-	if options != nil {
-		cp = *options
+	if options == nil {
+		options = &arm.ClientOptions{}
 	}
-	if len(cp.Endpoint) == 0 {
-		cp.Endpoint = arm.AzurePublicCloud
+	ep := string(options.Endpoint)
```

Comment:
Probably remove the cast here then just cast when assigning to host like before.